File: source/math/Sphere.java
package math;

import java.io.Serializable;

public class Sphere implements Serializable {
    public Float3 center;
    public float radius;

    public Sphere(Float3 center, float radius) {
        this.center = center;
        this.radius = radius;
    }

    public Sphere(Sphere sphere) {
        center = new Float3(sphere.center);
        radius = sphere.radius;
    }
}

File: source/math/Sphere.kt
package math

import java.io.Serializable

class Sphere : Serializable {
    var center: Float3 = Float3()
    var radius: Float = 0f

    constructor()

    constructor(center: Float3, radius: Float) {
        this.center = center
        this.radius = radius
    }

    constructor(sphere: Sphere) {
        center = Float3(sphere.center)
        radius = sphere.radius
    }
}